lib/barbeque/config.rb

```diff
@@ -0,0 +1,43 @@
+require 'erb'
+require 'yaml'
+
+module Barbeque
+  class Config
+    attr_accessor :exception_handler, :runner, :runner_options
+
+    def initialize(options = {})
+      options.each do |key, value|
+        if respond_to?("#{key}=")
+          public_send("#{key}=", value)
+        else
+          raise KeyError.new("Unexpected option '#{key}' was specified.")
+        end
+      end
+      init_runner_options
+    end
+
+    private
+
+    def init_runner_options
+      self.runner_options ||= {}
+      self.runner_options.symbolize_keys!
+    end
+  end
+
+  module ConfigBuilder
+    DEFAULT_CONFIG = {
+      'exception_handler' => 'RailsLogger',
+      'runner'            => 'Docker',
```

Comment:
[nits] The default value for `runner_options` should be given here instead of `init_runner_options`